error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/BigDecimalRules.java

```diff
@@ -66,4 +66,114 @@ BigDecimal after(double value) {
       return BigDecimal.valueOf(value);
     }
   }
+
+  /** Prefer using {@link BigDecimal#signum()} over comparing to {@link BigDecimal#ZERO}. */
+  static final class BigDecimalCompareToZero {
+    @BeforeTemplate
+    boolean before(BigDecimal value) {
+      return Refaster.anyOf(
+          value.compareTo(BigDecimal.ZERO) == 0, BigDecimal.ZERO.compareTo(value) == 0);
+    }
+
+    @AfterTemplate
+    boolean after(BigDecimal value) {
+      return value.signum() == 0;
+    }
+  }
+
+  /** Prefer using {@link BigDecimal#signum()} over comparing to {@link BigDecimal#ZERO}. */
+  static final class BigDecimalCompareToNonZero {
```

Comment:
Ah, we can use the `@AlsoNegation` and drop this template and get the same result :smile:.
We can move over the test cases though.